### ISSUE TITLE
No CPU Profiling option

### ISSUE THREAD

**Issue by arnsa** [facebook/react-native#12679]
Hello,

I'm developing an app using RN and I wanted to make some performance tests, but I can't find "Start CPU Profiling" option in dev menu on iOS. It's just not there. Why?

**Comment by AlanFoster** [comment]
@arnsa Not sure if you found out your answer or not; But based on https://youtu.be/0MlT74erp60?t=11m44s you have to compile your own dynamic library for your specific iOS simulator

**Comment by Sam1301** [comment]
@AlanFoster could you please elaborate more on how to create the library for a specific iOS simulator using  `new-library` command? I couldn't find anything in the `react-native --help` or the docs.

**Comment by nerdmed** [comment]
@tadeuzagallo would be great to leave a comment on how to activate the CPU Profiling we cant find any information on that. And your presentation is more then one year ago.

**Comment by tadeuzagallo** [comment]
Yeah, it's been a while since I stopped working on that. Apparently everything is still there, you can try running it by compiling the CPU profiler support library. All you have to do is run `make ios9` inside the folder `JSCLegacyProfiler`, then just recompile your app and the option should now be in the dev menu. If that doesn't work, I honestly don't know, it's been too long.

**Comment by nerdmed** [comment]
@tadeuzagallo thanks for your response. I needed to change the url inside the makefile and i have downloaded an old ios8 sdk as promted. But it still failes with the following error:

```
/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator80.sdk \
		`find download/JavaScriptCore/JavaScriptCore-7600.1.17 -name '*.h' | xargs -I{} dirname {} | uniq | xargs -I{} echo "-I {}"` \
		-undefined dynamic_lookup \
		JSCLegacyProfiler.mm
In file included from <built-in>:1:
In file included from ././download/JavaScriptCore/JavaScriptCore-7600.1.17/config.h:56:
In file included from download/WTF/WTF-7600.1.24/wtf/FastMalloc.h:25:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator80.sdk/usr/include/stdlib.h:63:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator80.sdk/usr/include/_types.h:27:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator80.sdk/usr/include/sys/_types.h:32:
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator80.sdk/usr/include/sys/cdefs.h:680:2: error:
      Unsupported architecture
#error Unsupported architecture
 ^
```

**Comment by michaelknoch** [comment]
`Makefile.base:31: *** "Expected to be compiled with iOS SDK version 8, found 10".  Stop.`
do you have a recommendation how to fix? @tadeuzagallo

**Comment by nerdmed** [comment]
@tadeuzagallo this does not seem to lead to anything that will work. Total waste of time. I am wondering why this is still inside the react-native repo. Do you know if there could be anyone else in the RN team that might be interested in adding a CPU profiling to React Native?

**Comment by tadeuzagallo** [comment]
> Total waste of time.

Sorry for wasting your time, sir. I shall aim to do better next time I'm helping someone on my personal time.

> I am wondering why this is still inside the react-native repo.

All I can suggest is that you send a pull request that removes it and see what the team will say.

> Do you know if there could be anyone else in the RN team that might be interested in adding a CPU profiling to React Native?

The version of the CPU profiler that was used is this script has been removed from recent versions of JavaScriptCore. I'd check if there's any better tooling built into Safari for debugging JSC on iOS by now. I'd think that's the most promising option. However, I'm *not* part of the team anymore, so I'm probably not the best person to answer that question.

**Comment by nerdmed** [comment]
@tadeuzagallo i really did not mean your work by waste of time, just wanted to prevent other developers to build the dylib and be disappointed. I think your investment in a cpu profiling tool is brilliant. Like you have presented in your video: Its the only thing that really makes sense when debugging performance issues. Its sad to see that no one in the team took the effort to keep this important and still missing peace working and well documented. It would be great to get a comment about this from the RN team before investing in a big PR. I found some evidence in the source code that someone is working on the new `JSCSamplingProfiler` ?!

**Comment by tadeuzagallo** [comment]
@nerdmed The sampling profiler replaced the LegacyProfiler in recent versions of JSC, however neither of them export a proper API. However, we managed to call the legacy profiler without modifying JSC, but I don't think that's possible with the sampling profiler. My knowledge might be out of date though, as it's been a while since I stopped working on it. I don't who would be the best person to talk to about this, but maybe @ndfred can point you in the right direction?!
